src/stream-tcp-reassemble.c

```diff
@@ -728,10 +728,13 @@ int StreamTcpReassembleHandleSegmentHandleData(ThreadVars *tv, TcpReassemblyThre
 
     /* If we have reached the defined depth for either of the stream, then stop
        reassembling the TCP session */
+    bool depth_reached = (stream->flags & STREAMTCP_STREAM_FLAG_DEPTH_REACHED);
     uint32_t size = StreamTcpReassembleCheckDepth(ssn, stream, TCP_GET_SEQ(p), p->payload_len);
     SCLogDebug("ssn %p: check depth returned %"PRIu32, ssn, size);
 
-    if (stream->flags & STREAMTCP_STREAM_FLAG_DEPTH_REACHED) {
+    /* Only count/event when the depth is first reached */
+    if (stream->flags & STREAMTCP_STREAM_FLAG_DEPTH_REACHED && !depth_reached) {
```

Comment:
I don't get it, why are we doing this depth_reached check if we should only be called once? See my last comment in the prev PR

Reply:
I misread your comment. Update coming.